mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -0,0 +1,70 @@
+import os
+import requests
+import posixpath
+
+from mlflow.entities import FileInfo
+from mlflow.store.artifact.artifact_repo import ArtifactRepository, verify_artifact_path
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+
+
+class HttpArtifactRepository(ArtifactRepository):
+    """Stores artifacts in a remote artifact storage using HTTP requests"""
+
+    def __init__(self, artifact_uri):
+        super().__init__(artifact_uri)
+        self._session = requests.Session()
```

Comment:
Replaced `request.xxx` to with `self._session.xxx` to avoid re-establishing a new connection everytime we make a request.

Reply:
neat!